modules/openy_features/openy_location/modules/openy_loc_camp/src/CampHelper.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+namespace Drupal\openy_loc_camp;
+
+use Drupal\node\Entity\Node;
+use Drupal\node\NodeInterface;
+
+/**
+ * Class Camp Helper functions.
+ */
+class CampHelper {
+
+  /**
+   * Helper that returns the Camp CT associated to the landing page provided.
+   *
+   * @param \Drupal\openy_prgf_camp_menu\NodeInterface $node
+   *   The node.
+   */
+  public static function getLandingPageCampNode(NodeInterface $node) {
+    $camp = NULL;
+    if ($node->bundle() != 'landing_page') {
+      return $camp;
+    }
+
+    if ($node->hasField('field_location')) {
+      if ($value = $node->field_location->referencedEntities()) {
+        $camp = reset($value);
+      }
+    }
+    // Else if a camp links to this landing page use the linking camp.
+    else {
+      // Query 1 Camp nodes that link to this landing page.
+      $query = \Drupal::entityQuery('node')
+        ->condition('status', 1)
+        ->condition('type', 'camp')
```

Comment:
can we avoid of adding CTs to the conditions? field level could be enough, I guess